util/flags/manager.go

```diff
@@ -0,0 +1,144 @@
+*/
+
+// Package flags implements the manager options utilities.
+package flags
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apiserver/pkg/server/routes"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/component-base/logs"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/filters"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+// ManagerOptions provides command line flags for manager options.
+type ManagerOptions struct {
+	// TLS Options
+	// These are used for the webhook and for the metrics server (the latter only if secure serving is used)
+
+	TLSMinVersion   string
+	TLSCipherSuites []string
+
+	// Metrics Options
+	// These are used to configure the metrics server
+
+	// MetricsBindAddr
+	//
+	// Deprecated: This field will be removed in an upcoming release.
+	MetricsBindAddr     string
+	DiagnosticsAddress  string
+	InsecureDiagnostics bool
+}
+
+// AddManagerOptions adds the manager options flags to the flag set.
+func AddManagerOptions(fs *pflag.FlagSet, options *ManagerOptions) {
+	fs.StringVar(&options.TLSMinVersion, "tls-min-version", "VersionTLS12",
+		"The minimum TLS version in use by the webhook server.\n"+
+			fmt.Sprintf("Possible values are %s.", strings.Join(cliflag.TLSPossibleVersions(), ", ")),
+	)
+
+	tlsCipherPreferredValues := cliflag.PreferredTLSCipherNames()
+	tlsCipherInsecureValues := cliflag.InsecureTLSCipherNames()
+	fs.StringSliceVar(&options.TLSCipherSuites, "tls-cipher-suites", []string{},
+		"Comma-separated list of cipher suites for the webhook server. "+
+			"If omitted, the default Go cipher suites will be used. \n"+
+			"Preferred values: "+strings.Join(tlsCipherPreferredValues, ", ")+". \n"+
+			"Insecure values: "+strings.Join(tlsCipherInsecureValues, ", ")+".")
+
+	fs.StringVar(&options.MetricsBindAddr, "metrics-bind-addr", "",
+		"The address the metrics endpoint binds to.")
+	_ = fs.MarkDeprecated("metrics-bind-addr", "Please use --diagnostics-address instead. To continue to serve"+
+		"metrics via http and without authentication/authorization set --insecure-diagnostics as well.")
+
+	fs.StringVar(&options.DiagnosticsAddress, "diagnostics-address", ":8443",
+		"The address the diagnostics endpoint binds to. Per default metrics are served via https and with"+
+			"authentication/authorization. To serve via http and without authentication/authorization set --insecure-diagnostics."+
+			"If --insecure-diagnostics is not set the diagnostics endpoint also serves pprof endpoints and an endpoint to change the log level.")
+
+	fs.BoolVar(&options.InsecureDiagnostics, "insecure-diagnostics", false,
+		"Enable insecure diagnostics serving. For more details see the description of --diagnostics-address.")
+}
+
+// GetManagerOptions returns options which can be used to configure a Manager.
+func GetManagerOptions(options ManagerOptions) ([]func(config *tls.Config), *metricsserver.Options, error) {
```

Comment:
it seems like a function that should be unit tested?

for example, the second return argument can never be nil, but it might be worth adding a unit test for that to avoid a panic on runtime.

Reply:
Added unit tests. Specifically validating that metricsOptions is never nil if err is nil